pkg/render/tiers/tiers.go

```diff
@@ -0,0 +1,333 @@
+			Selector: dnsPolicySelector,
+			Ingress: []v3.Rule{
+				{
+					Action: v3.Allow,
+					Source: v3.EntityRule{
+						NamespaceSelector: "all()",
+						Selector:          DNSIngressNamespaceSelector,
+					},
+				},
+				{
+					Action: v3.Pass,
+				},
+			},
+			Egress: []v3.Rule{
+				{
+					Action: v3.Allow,
+				},
+			},
+			Types: []v3.PolicyType{v3.PolicyTypeIngress, v3.PolicyTypeEgress},
+		},
+	}
+}
+
+// Allow the Kubernetes API Server access to Calico Enterprise API Server.
+// Reconciled here rather than the API Server controller since the creation of v3 NetworkPolicy depends on the API Server.
+func (t *tiersComponent) allowTigeraAPIServerPolicy() *v3.NetworkPolicy {
+	egressRules := []v3.Rule{}
+	egressRules = networkpolicy.AppendDNSEgressRules(egressRules, t.cfg.Openshift)
+	egressRules = append(egressRules, []v3.Rule{
+		{
+			Action:      v3.Allow,
+			Protocol:    &networkpolicy.TCPProtocol,
+			Destination: networkpolicy.KubeAPIServerEntityRule,
+		},
+		{
+			// Pass to subsequent tiers for further enforcement
+			Action: v3.Pass,
+		},
+	}...)
+
+	// The ports Calico Enterprise API Server and Calico Enterprise Query Server are configured to listen on.
+	ingressPorts := networkpolicy.Ports(443, render.ApiServerPort, render.QueryServerPort, 10443)
+
+	return &v3.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "projectcalico.org/v3"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      APIServerPolicyName,
+			Namespace: rmeta.APIServerNamespace(operatorv1.TigeraSecureEnterprise),
+		},
+		Spec: v3.NetworkPolicySpec{
+			Order:    &networkpolicy.HighPrecedenceOrder,
+			Tier:     networkpolicy.TigeraComponentTierName,
+			Selector: networkpolicy.KubernetesAppSelector("tigera-apiserver"),
+			Types:    []v3.PolicyType{v3.PolicyTypeIngress, v3.PolicyTypeEgress},
+			Ingress: []v3.Rule{
+				{
+					Action:   v3.Allow,
+					Protocol: &networkpolicy.TCPProtocol,
+					// This policy allows Calico Enterprise API Server access from anywhere.
+					Source: v3.EntityRule{
+						Nets: []string{"0.0.0.0/0"},
+					},
+					Destination: v3.EntityRule{
+						Ports: ingressPorts,
+					},
+				},
+				{
+					Action:   v3.Allow,
+					Protocol: &networkpolicy.TCPProtocol,
+					Source: v3.EntityRule{
+						Nets: []string{"::/0"},
+					},
+					Destination: v3.EntityRule{
+						Ports: ingressPorts,
+					},
+				},
+			},
+			Egress: egressRules,
+		},
+	}
+}
+
+// Reconciled here rather than the core controller since the creation of v3 NetworkPolicy depends on the API Server, and
+// the core controller is a prerequisite of the API server controller.
+func (t *tiersComponent) allowTigeraKubeControllersPolicy() *v3.NetworkPolicy {
+	egressRules := []v3.Rule{}
+	egressRules = networkpolicy.AppendDNSEgressRules(egressRules, t.cfg.Openshift)
+	egressRules = append(egressRules, []v3.Rule{
+		{
+			Action:   v3.Allow,
+			Protocol: &networkpolicy.TCPProtocol,
+			Destination: v3.EntityRule{
```

Comment:
This doesn't use the 'shared' API Server entity rule since it doesn't specify a namespace selector like all other policies do.

Is there a reason why this is the case, or can I align this with the others?